### PR TITLE
Feat/grouping GET/PUT

### PR DESCRIPTION
**Pull request by sellinjaanus**
Added 2 logics.

Getting a single group.
Updating a single group

### CODE REVIEW

File: src/lib/services/group-service.ts [hunk 1]
```diff
@@ -35,9 +35,9 @@ export class GroupService {
         const users = await this.userStore.getAllWithId(
             allGroupUsers.map((u) => u.userId),
         );
-        return groups.map((a) => {
+        return groups.map((group) => {
```
Comment (sighphyre) [inline src/lib/services/group-service.ts:38]: Nice thank you!

File: src/lib/services/group-service.ts [hunk 2]
```diff
@@ -71,14 +84,58 @@ export class GroupService {
...
+            userName,
+        );
+
+        const deletableUsers = existingUsers.filter(
+            (existingUser) =>
+                !group.users.some(
+                    (groupUser) => groupUser.user.id == existingUser.userId,
+                ),
+        );
+        await this.groupStore.deleteOldUsersFromGroup(deletableUsers);
```
Comment (sighphyre) [inline src/lib/services/group-service.ts:113]: There's a lot of queries going on in this function. I think mostly this is fine but I do feel that we should wrap the remove users and add new users into a transaction if possible. Probably not going to be a problem but if there ever is a race condition that leaves groups in a weird state this is going to be horrid to debug

**Review by sighphyre — APPROVED** [review]
Looks good overall, I think the conversation about the transaction can be handled in a different PR (or tell I'm crazy)

File: src/lib/services/group-service.ts (hunk 2 above)
Comment (sellinjaanus) [reply src/lib/services/group-service.ts:113]: Agreed totally. I will do it in new pull request.

**Pull request merged**